### PR TITLE
feat: new operators `every` and `some`

### PR DESCRIPTION
**Pull request by RebeccaStevens**

### CODE REVIEW

**Comment by tim-smart** [comment]
Thanks for the contribution!

However, I think I would like to avoid using `toAsyncIterable` in operators, as it adds unnecessary overhead.

There is an internal `subscribe` helper you can use, to make it more callbag-ish. `lastItemFrom` uses it as an example.

**Comment by RebeccaStevens** [comment]
Updated :)

File: api/every.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    new Promise((resolve, reject) => {
+      let result = true
+      let index = 0
+
+      const sub = subscribe(self, {
+        onStart() {
+          sub.pull()
+        },
+        onData(data) {
+          result = result && pred(data, index++)
```
Comment (tim-smart) [inline api/every.ts:24]: Here you can add a condition guarding a call to `sub.cancel()`.

File: api/some.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    new Promise((resolve, reject) => {
+      let result = false
+      let index = 0
+
+      const sub = subscribe(self, {
+        onStart() {
+          sub.pull()
+        },
+        onData(data) {
+          result = result || pred(data, index++)
```
Comment (tim-smart) [inline api/some.ts:24]: If an element passes, then`sub.cancel()`

**Review by tim-smart — CHANGES_REQUESTED** [review]

**Comment by RebeccaStevens** [comment]
fix submitted

**Comment by tim-smart** [comment]
Thanks - I just made a quick tweak to fix the failing tests :)

**Comment by RebeccaStevens** [comment]
Opps, I should have tested before pushing :P

**Pull request merged**